packages/diagram-ui/src/features/update/updateModel.ts

```diff
@@ -109,6 +115,70 @@ export class UpdateModelCommand extends BaseUpdateModelCommand {
     ): CommandReturn {
         (newRoot as SRoot).changeRevision = (oldRoot as SRoot).changeRevision + 1;
         (newRoot as SRoot).sequenceNumber = (oldRoot as SRoot).sequenceNumber;
+        layoutViewportIfNecessary(oldRoot.index.all(), newRoot.index.all(), this.dispatcher);
         return super.performUpdate(oldRoot, newRoot, context);
     }
 }
+
+/**
+ * Fits the viewport to the new diagram size if too many changes occurred (i.e. the user pasted half a diagram into the editor).
+ *
+ * @param newElements the index storing all new model elements
+ * @param old the index storing all current model elements
+ * @param dispatcher the dispatcher to dispatch the 'fit to screen' action
+ */
+function layoutViewportIfNecessary(
```

Comment:
why not move into the command? then you don't need to pass the action dispatcher

Reply:
I don't think this belongs into the command execution inherently.
Instead, it is its own thing that happens to be called inside the command execution.

Reply:
well, as long as the docs are there, I don't care...